pkg/serving/config_changes.go

```diff
@@ -269,3 +344,200 @@ func removeEnvVars(env []corev1.EnvVar, toRemove []string) []corev1.EnvVar {
 	}
 	return env
 }
+
+func parseVolumeSource(s string) (VolumeSourceType, string, error) {
+	slices := strings.SplitN(s, ":", 2)
+	if len(slices) != 2 {
+		return -1, "", fmt.Errorf("Argument requires a value that contains the : character; got %q", s)
+	}
+
+	if strings.TrimSpace(slices[0]) == "config-map" {
+		return ConfigMapVolumeSourceType, strings.TrimSpace(slices[1]), nil
+	} else if strings.TrimSpace(slices[0]) == "secret" {
+		return SecretVolumeSourceType, strings.TrimSpace(slices[1]), nil
+	}
+
+	return -1, "", fmt.Errorf("Not allowed key-value source is used. Currently config-map and secret can be used; got %q", s)
```

Comment:
"Unsupported volume source type "asdf"; supported volume source types are "config-map" and "secret"

Reply:
It will be changed as your comments